src/bsoncxx/CMakeLists.txt

```diff
@@ -21,7 +21,13 @@ set(BSONCXX_HEADER_INSTALL_DIR "include/bsoncxx/${BSONCXX_INLINE_NAMESPACE}" CAC
 option(BSONCXX_POLY_USE_STD_EXPERIMENTAL "Use <experimental/ for stdx polyfills" OFF)
 option(BSONCXX_POLY_USE_MNMLSTC "Use mnmlstc/core for stdx polyfills" ON)
 option(BSONCXX_POLY_USE_SYSTEM_MNMLSTC "Obtain mnmlstc/core from system" OFF)
-option(BSONCXX_POLY_USE_BOOST "Use boost for stdx polyfills" OFF)
+
+# MNMLSTC is broken on the OSX evergreen builder
+if(CMAKE_OSX)
+  option(BSONCXX_POLY_USE_BOOST "Use boost for stdx polyfills" ON)
```

Comment:
I think it might be better to pass in `-DBSONCXX_POLY_USE_BOOST` from our evergreen script on the osx variants rather than change this here